instrumentation/aws-sdk/aws-sdk-1.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/awssdk/v1_11/TracingRequestHandler.java

```diff
@@ -27,19 +34,42 @@ public TracingRequestHandler(ContextStore<AmazonWebServiceRequest, RequestMeta>
 
   @Override
   public void beforeRequest(Request<?> request) {
+
     AmazonWebServiceRequest originalRequest = request.getOriginalRequest();
+    Span.Kind kind = (isSqsProducer(originalRequest) ? Span.Kind.PRODUCER : Span.Kind.CLIENT);
+
     RequestMeta requestMeta = contextStore.get(originalRequest);
     Context parentContext = Context.current();
     if (!tracer().shouldStartSpan(parentContext)) {
       return;
     }
-    Context context = tracer().startSpan(parentContext, request, requestMeta);
+    Context context = tracer().startSpan(kind, parentContext, request, requestMeta);
     Scope scope = context.makeCurrent();
     request.addHandlerContext(CONTEXT_SCOPE_PAIR_CONTEXT_KEY, new ContextScopePair(context, scope));
+    // enforce AWS propagation
```

Comment:
Sorry for extracting these into #2117 hopefully a simple rebase

Reply:
:D